action/protocol/staking/candidate_center.go

```diff
@@ -17,7 +17,8 @@ import (
 type (
 	// candChange captures the change to candidates
 	candChange struct {
-		dirty map[string]*Candidate
+		candidates []*Candidate
+		dirty      map[string]*Candidate
```

Comment:
we should be able to do the fix w/o modifying these low-level implementations

Reply:
could either `map` or `list` be used in the `candChange` struct?